### PR TITLE
[Returning Customers] Introduce Customer Authorization

### PR DESCRIPTION
**Pull request by mikeclimbrock**
Conceptually, allow resource authorization by `Bearer` JWT, where the claim set includes a `session id` and `customer number`.

In practice, `proxy` will be updated to include a new parser that generates an `Authorization.Customer`.
1) A `Customer` auth is built with the session and customer number parsed from the clam set.
2) A `ResolvedToken` is built with the organization id/environment obtained from the session authorization (as is the case today for session authentication)

Note that the `Authorization` controller in this lib has intentionally **not** been updated to parse a new JWT model.  This controller is not utilized by `proxy` (proxy has it's own lib), so updating is not necessary at this time.

`lib-play` simply receives a `data` map as set by `proxy`, thus, resources authorized with the "wrapper" `Org` (`OrgAuthData`), will now include a step that attempts to authenticate as a `Customer`.

For completeness and transparency, `customer` (optional) has been threaded through the other authorization models.

### CODE REVIEW

File: app/io/flow/play/controllers/FlowRequests.scala [hunk 1]
```diff
@@ -43,6 +43,16 @@ class IdentifiedOrgRequest[A](
   val environment: Environment = auth.environment
 }
 
+class CustomerRequest[A](
```
Comment (jlogeart) [inline app/io/flow/play/controllers/FlowRequests.scala]: let's fix indentation here

Comment (mikeclimbrock) [reply app/io/flow/play/controllers/FlowRequests.scala]: originally, made it consistent.  instead, have update all indentation

File: app/io/flow/play/controllers/FlowController.scala [hunk 2]
```diff
@@ -142,3 +145,14 @@ object IdentifiedCookie {
   }
 
 }
+
+// Customer
+class CustomerActionBuilder @Inject()(val parser: BodyParsers.Default, val config: Config, implicit private val logger: RollbarLogger)(implicit val executionContext: ExecutionContext)
```
Comment (jlogeart) [inline app/io/flow/play/controllers/FlowController.scala]: let's move all implicits to the second parameters

File: app/io/flow/play/controllers/FlowRequests.scala (hunk 1 above)
Comment (jlogeart) [reply app/io/flow/play/controllers/FlowRequests.scala]: Ah it did not show in the diff :)

File: app/io/flow/play/controllers/FlowController.scala (hunk 2 above)
Comment (mikeclimbrock) [reply app/io/flow/play/controllers/FlowController.scala]: sure - updated

**Comment by jlogeart** [comment]
LGTM.

**Pull request merged**